### PR TITLE
DOC: Clean .ix references in indexing.rst

### PR DESCRIPTION
**Pull request by topper-123**
`DataFrame.ix` is removed from the code base, so rephrase a section in the docs that was referencing it.

### CODE REVIEW

File: doc/source/user_guide/indexing.rst [hunk 1]
```diff
@@ -584,45 +584,13 @@ without using a temporary variable.
    (bb.groupby(['year', 'team']).sum()
       .loc[lambda df: df['r'] > 100])
 
-.. _indexing.deprecate_ix:
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/indexing.rst:587]: There is one reference to this label in v0.20.0.rst. Could you replace that with a harcoded link to an older version of the docs? Eg https://pandas.pydata.org/pandas-docs/version/1.0/user_guide/indexing.html#ix-indexer-is-deprecated

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks!

It seems you have picked another commit

File: doc/source/whatsnew/v1.2.0.rst [hunk 2]
```diff
@@ -627,6 +627,7 @@ Indexing
 - Bug in :meth:`DataFrame.loc` raising ``TypeError`` when non-integer slice was given to select values from :class:`MultiIndex` (:issue:`25165`, :issue:`24263`)
 - Bug in :meth:`DataFrame.loc` returning and assigning elements in wrong order when indexer is differently ordered than the :class:`MultiIndex` to filter (:issue:`31330`, :issue:`34603`)
 - Bug in :meth:`DataFrame.loc` and :meth:`DataFrame.__getitem__`  raising ``KeyError`` when columns were :class:`MultiIndex` with only one level (:issue:`29749`)
+- Bug in :meth:`Series.__getitem__` and :meth:`DataFrame.__getitem__` raising blank ``KeyError`` without missing keys for :class:`IntervalIndex` (:issue:`27365`)
```
Comment (phofl) [inline doc/source/whatsnew/v1.2.0.rst:630]: Weird that this is mentioned as change. That is from one of mine PRs which was merged a few hours ago

File: doc/source/user_guide/indexing.rst (hunk 1 above)
Comment (topper-123) [reply doc/source/user_guide/indexing.rst:587]: Done.

File: doc/source/whatsnew/v1.2.0.rst (hunk 2 above)
Comment (topper-123) [reply doc/source/whatsnew/v1.2.0.rst]: This was a rebasing bug. I've fixed it now.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jreback** [comment]
thanks @topper-123 nice

**Pull request merged**